iocore/net/SSLConfig.cc

```diff
@@ -717,12 +731,22 @@ SSLConfigParams::getCTX(const char *client_cert, const char *key_file, const cha
   }
 
   ink_mutex_acquire(&ctxMapLock);
-  iter = ctx_map.find(key);
-  if (iter != ctx_map.end()) {
+  top_iter = top_level_ctx_map.find(top_level_key);
+  if (top_iter != top_level_ctx_map.end()) {
+    if (top_iter->second == nullptr) {
+      top_iter->second = new CTX_MAP;
+    }
+    ctx_map = top_iter->second;
+  } else {
+    ctx_map = new CTX_MAP;
+    top_level_ctx_map.insert(std::make_pair(top_level_key, ctx_map));
+  }
+  iter = ctx_map->find(ctx_key);
+  if (iter != ctx_map->end()) {
     SSL_CTX_free(client_ctx);
     client_ctx = iter->second;
   } else {
-    ctx_map.insert(std::make_pair(key, client_ctx));
+    ctx_map->insert(std::make_pair(ctx_key, client_ctx));
```

Comment:
This insert could be fail, when this function do race of creating and inserting `client_ctx`. Because the lock is released after the first lookup.
And when this insert is failed, it looks like the `client_ctx` is not freed.

Reply:
On line 744, it is checking for existence of ctx_key with the lock held. And if such a key already exists (due to race of creating and inserting, it will release local client_ctx and set the pointer to the one in ctx_map.

Reply:
I missed that line. Thanks for explanations!